Dockerfile

```diff
@@ -17,7 +17,9 @@ COPY pkg/ pkg/
 
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+# Default the GOARCH to amd64, but allow for overide to a different arch at build time
+ARG ARCH=amd64
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} go build -a -o manager main.go
```

Comment:
I don't think we need to define the GOARCH here at all. I would imagine buildx's --platform should override this in any case.

So this should be defined in the Makefile for now to allow overriding the target platform (target docker-build)

Reply:
I'll try it without the GOARCH.  `buildx` didn't override it, I ended up with an arm64 image with an amd64 executable.  But without GOARCH at all, it might work.

I'll update here and change the PR if it works.

Reply:
I think without GOARCH it should work, I quickly tested on a Ampere Altra instance:

```
.PHONY: docker-build
docker-build: ## Build docker image with the manager.
	docker buildx build --build-arg VERSION=${VERSION} -t ${IMG} -t ${IMG_LATEST} . --load --platform linux/arm64/v8
```

It built (with docker-kind):

```
k8ssandra/cass-operator           v1.14.0-dev.f42c450-20221028   bb42848f8837   10 minutes ago   76.5MB
```

And after ``make cert-manager deploy`` I have running system:

```
NAME                                                READY   STATUS    RESTARTS   AGE
cass-operator-controller-manager-77bf465f7c-r24f2   1/1     Running   0          2m17s
```

The system:

```
➜  cass-operator git:(master) ✗ uname -a
Linux instance-20210812-2349 5.19.0-1010-oracle #11-Ubuntu SMP Mon Oct 17 01:43:14 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
➜  cass-operator git:(master) ✗ 
```

Reply:
And ye, the Dockerfile modification was:

```
# Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux go build -a -o manager main.go
```

Reply:
Note that there's a small limitation in that we can't build multiple --platforms if --load is used. But I'm not sure if the docker-build command even needs the --platform target in this case.

Reply:
Building with `docker buildx build --platform=linux/arm64/v8 --load -t docker.io/kentbull/k8ssandra-operator:arm64 .` worked for me. I did have to modify the Dockerfile as you did @burmanm.